Update patches for polkadot 0.9.43

fairax opened the pull request.

Commit:
feat: make OnChargeEvmTransaction aware of user action

Commit:
feat: sponsorship handler

CertainLach committed:
feat: fake transactions

Allow outer code to inject ethereum logs, without running an ethereum
transaction.

CertainLach committed:
hack: special charging logic on call and estimate

Commit:
feat: use storage for pending logs

Allow outer code to inject logs for currently running ethereum
transaction.

Commit:
feat: extended precompile interface

Commit:
feat: transfer `value` from substrate account if called with `evm.call`

fairax commented on client/db/Cargo.toml:
```diff
@@ -45,6 +45,8 @@ tempfile = "3.3.0"
 sp-consensus = { workspace = true }
 substrate-test-runtime-client = { workspace = true }
+# Frontier
+frontier-template-runtime = { workspace = true, features = ["default"] }
```
Добавил чтобы починить проблему с новыми тестами из `[sql backend w/ receipt log indexing](https://github.com/paritytech/frontier/pull/883/files#top)`

CertainLach replied:
Я не вижу в тестах использования рантайма, можно подробнее?

CertainLach replied:
Вернее, я вижу
let (client, _) =
			builder.build_with_native_executor::<frontier_template_runtime::RuntimeApi, _>(None);
Но не понимаю зачем нам там использовать именно рантайм

fairax replied:
У нас добавились констреинты для `SchemaV3Override`
`C: ProvideRuntimeApi<B>,
C::Api: EthereumRuntimeRPCApi<B>`
Чтобы они выполнялись понадобился другой рантайм.

fairax commented on frame/evm/src/runner/stack.rs:
```diff
@@ -1199,6 +1383,80 @@ where
+			is_precompile: true,
+			..
+		} = result
+		{
```
Переделал метод из-за изменений в прекомпайлах

CertainLach commented on frame/evm/precompile/dispatch/src/mock.rs:
```diff
@@ -93,7 +93,7 @@ impl pallet_utility::Config for Test {
 
 parameter_types! {
-	pub const ExistentialDeposit: u64 = 0;
+	pub const ExistentialDeposit: u64 = 1;
```
А зачем? Разве в тестах не включён insecure_zero_ed?

fairax replied:
Это уже было в патчах. Можно попробовать переделать.

fairax replied:
Убрал

CertainLach commented on frame/ethereum/src/mock.rs:
```diff
@@ -158,14 +158,37 @@ impl AddressMapping<AccountId32> for HashedAddressMapping {
+	fn into_account_id(account_id: H160) -> AccountId32 {
+		let mut data = [0u8; 32];
+		data[0..20].copy_from_slice(account_id.as_bytes());
+		AccountId32::from(data)
```
Конверсия из эфира в субстрат каноничная это HashedAddressMapping.
Я не вижу где бы мы пользовались этим конвертером, но его в любом случае стоит убрать как некорректный

CertainLach commented:
Последние 2 патча (value + fix transfer) стоит засквошить/заfixupить
С первыми двумя (версия+фиксы) нужно поступить аналогично

Commit:
feat: move address conversions to the lower level

1. Allows to pay for transactions performed with evm.call using balance
   of source substrate address.
2. Same applies to later introduced sponsoring.
3. Definition of CrossAccountId enum, for substrate-ethereum
   intercompatibility.
